packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts

```diff
@@ -1,69 +1,38 @@
+import { GRID_ROW_CSS_CLASS } from '../../../constants/cssClassesConstants';
 import {
+  GRID_CELL_KEYDOWN,
+  GRID_CELL_NAVIGATION_KEYDOWN,
   GRID_ELEMENT_FOCUS_OUT,
   GRID_KEYDOWN,
   GRID_KEYUP,
   GRID_MULTIPLE_KEY_PRESS_CHANGED,
 } from '../../../constants/eventsConstants';
 import { GridApiRef } from '../../../models/api/gridApiRef';
-import { GridCellIndexCoordinates } from '../../../models/gridCell';
+import { GridCellParams } from '../../../models/params/gridCellParams';
 import {
   findParentElementFromClassName,
   getIdFromRowElem,
   getRowEl,
   isGridCellRoot,
 } from '../../../utils/domUtils';
-import {
-  isArrowKeys,
-  isHomeOrEndKeys,
-  isMultipleKey,
-  isNavigationKey,
-  isPageKeys,
-  isSpaceKey,
-} from '../../../utils/keyboardUtils';
-import { optionsSelector } from '../../utils/optionsSelector';
-import { visibleGridColumnsLengthSelector } from '../columns/gridColumnsSelector';
+import { isMultipleKey, isNavigationKey, isSpaceKey } from '../../../utils/keyboardUtils';
 import { useGridSelector } from '../core/useGridSelector';
 import { useGridState } from '../core/useGridState';
-import { gridPaginationSelector } from '../pagination/gridPaginationSelector';
-import { gridRowCountSelector } from '../rows/gridRowsSelector';
 import { useLogger } from '../../utils/useLogger';
 import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
 import { gridSelectionStateSelector } from '../selection/gridSelectionSelector';
 import { KeyboardState } from './keyboardState';
-
-const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) => {
-  if (!isArrowKeys(code)) {
-    throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
-  }
-
-  if (code === 'ArrowLeft') {
-    return { ...indexes, colIndex: indexes.colIndex - 1 };
-  }
-  if (code === 'ArrowRight') {
-    return { ...indexes, colIndex: indexes.colIndex + 1 };
-  }
-  if (code === 'ArrowUp') {
-    return { ...indexes, rowIndex: indexes.rowIndex - 1 };
-  }
-  // Last option code === 'ArrowDown'
-  return { ...indexes, rowIndex: indexes.rowIndex + 1 };
-};
+import { useGridKeyboardNavigation } from './useGridKeyboardNavigation';
 
 export const useGridKeyboard = (
   gridRootRef: React.RefObject<HTMLDivElement>,
   apiRef: GridApiRef,
 ): void => {
   const logger = useLogger('useGridKeyboard');
-  const options = useGridSelector(apiRef, optionsSelector);
   const [, setGridState, forceUpdate] = useGridState(apiRef);
-  const paginationState = useGridSelector(apiRef, gridPaginationSelector);
-  const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
-  const colCount = useGridSelector(apiRef, visibleGridColumnsLengthSelector);
-  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
   const selectionState = useGridSelector(apiRef, gridSelectionStateSelector);
+  useGridKeyboardNavigation(gridRootRef, apiRef);
```

Comment:
I think it would be better if we move this to the `GridComponent` where all the hooks are being called and avoid nesting essential / feature hooks.

Reply:
put it there as it's part of the keyboard but yeah moved!